### DISCUSSION TITLE
PERF_COUNTERS UDP_RECV_EVENTS  and UDP_SEND_CALLS   meaning

### DISCUSSION THREAD

**Discussion by akrisfx** [microsoft/msquic#4713] [Q&A] [answered]
Hello again
PERF_COUNTERS has metrics 
```
QUIC_PERF_COUNTER_UDP_RECV_EVENTS // Total UDP receive events 
QUIC_PERF_COUNTER_UDP_SEND_CALLS // Total UDP send API calls
```
can we meaningfully equate them to `packets_received` and `packets_sended`?

**Comment by nibanks** [comment, marked as answer]
No, you can't.

- `QUIC_PERF_COUNTER_UDP_RECV_EVENTS` - This indicates how many receive callbacks QUIC gets from the platform. Each receive callback may have one or more UDP datagrams (and a single UDP datagram may have multiple QUIC packets).
- `QUIC_PERF_COUNTER_UDP_SEND_CALLS` - This is the number of send API calls the app makes. Each call may result in many packets being sent out.

I'm pretty sure you just want:

```c
typedef enum QUIC_PERFORMANCE_COUNTERS {
    ...
    QUIC_PERF_COUNTER_UDP_RECV,             // Total UDP datagrams received.
    QUIC_PERF_COUNTER_UDP_SEND,             // Total UDP datagrams sent.
```

**Reply by akrisfx** [reply]
I understand correctly that all the basic `send` calls by me in the application will be count in `QUIC_PERF_COUNTER_UDP_SEND_CALLS`?

**Reply by akrisfx** [reply]
I need to count the number of "my" internal app packets(1 proto message) that I have sent using QUIC
